src/sdc11073/location.py

```diff
@@ -56,55 +65,48 @@ def _mk_scope_string(self, elements):
         loc = slash.join(identifiers)  # this is a bit ugly, but urllib.quote does not touch slashes;
         query = urlencode(query_dict)
         path = f'/{quote(self.root)}/{loc}'
-        scope_string = urlunparse(
+        return urlunparse(
             ParseResult(scheme=self.scheme, netloc=None, path=path, params=None, query=query, fragment=None))
-        return scope_string
 
-    def matching_services(self, services):
+    def filter_services_inside(self, services: Iterable[Service]) -> list[Service]:
+        """Return services that are 'inside' own location."""
         return [s for s in services if self._service_matches(s)]
 
-    def _service_matches(self, service):
-        return self._any_scope_string_matches(service.scopes)
-
-    def _any_scope_string_matches(self, scopes):
-        for scope in scopes:
-            if self._scope_string_matches(scope):
-                return True
-        return False
+    def _service_matches(self, service: Service) -> bool:
+        if service.scopes is None:
+            return False
+        return any(self._scope_string_matches(scope) for scope in service.scopes.text)
 
-    def _scope_string_matches(self, scope):
-        """
-        Check if location in scope is inside own location.
-        :param scope: url string
-        :return: boolean
-        """
+    def _scope_string_matches(self, scope_text: str) -> bool:
+        """Check if location in scope is inside own location."""
         try:
-            other = self.__class__.from_scope_string(str(scope))
+            other = self.__class__.from_scope_string(str(scope_text))
```

Comment:
no need to cast it to `str` as the parameter type indicates that it is already a string